code/client/makecatalogs

```diff
@@ -150,6 +199,29 @@ def makecatalogs(repopath, options):
                 if key.startswith('_'):
                     del(pkginfo[key])
 
+
+            if iconhashing:
+
+                name = pkginfo.get('name')
+
+                if pkginfo.get('icon_name'):
+                    try:
+                        iconhash = icons.get(icon_name)
+                    except:
+                        del(pkginfo['icon_name'])
```

Comment:
What if it logs a warning about the missing icon file similar to a missing installer item (in pkgs) is handled, and mentions that the key was simply ignored?